### PR TITLE
Do not check whether key == null in DKV.get

### PR DESCRIPTION
**Pull request by st-pasha**
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/169)

### CODE REVIEW

File: h2o-algos/src/main/java/hex/tree/SharedTreeModel.java [hunk 1]
```diff
@@ -347,9 +347,10 @@ protected void writeModelData() throws IOException {
       int ntreesPerClass = binomialOpt() && nclasses == 2? 1 : nclasses;
       for (int i = 0; i < _output._ntrees; i++) {
         for (int j = 0; j < ntreesPerClass; j++) {
-          Value ctVal = DKV.get(_output._treeKeys[i][j]);
+          Key<CompressedTree> key = _output._treeKeys[i][j];
+          Value ctVal = key != null? DKV.get(key) : null;
```
Comment (arnocandel) [inline h2o-algos/src/main/java/hex/tree/SharedTreeModel.java]: what would DKV.get(null) return? is this check still needed?

File: h2o-core/src/main/java/water/DKV.java [hunk 2]
```diff
@@ -185,7 +185,6 @@ static void write_barrier() {
   static public void prefetch( String key_name ) {        get(Key.make(key_name),false); }
 
   static private Value get( Key key, boolean blocking ) {
-    if (key == null) return null;
```
Comment (arnocandel) [inline h2o-core/src/main/java/water/DKV.java]: not sure about this - could have undesirable side-effects

Comment (mmalohlava) [inline h2o-core/src/main/java/water/DKV.java]: this was introduced by POJO commit - we look at that with Tomas, and asked pasha to remove the line.

Comment (arnocandel) [inline h2o-core/src/main/java/water/DKV.java]: ok thx

Comment (mmalohlava) [inline h2o-core/src/main/java/water/DKV.java]: the reasons are:
  - the method is private - the caller is DKV class itself, it has to handle null before calling this method 
 - performance impact - probably not big deal, but without proper perf test we can say nothing about  it
 - the commit was hidden in POJO commit - we need to be careful on changing core as a side effect of other changes, it is better to make explicit PR with core change

**Pull request merged**